### ISSUE TITLE
"Tried to spawn a new thread, but the thread pool is exhausted." spam

### ISSUE THREAD

**Issue by juj** [emscripten-core/emscripten#19353]
At some point Emscripten gained this error print in the logs:
```
Tried to spawn a new thread, but the thread pool is exhausted.
This might result in a deadlock unless some threads eventually exit or the code explicitly breaks out to the event loop.
If you want to increase the pool size, use setting `-sPTHREAD_POOL_SIZE=...`.
If you want to throw an explicit error instead of the risk of deadlocking in those cases, use setting `-sPTHREAD_POOL_SIZE_STRICT=2`.
```
However, in Unity's case, we are launching threads asynchronously. So we do not want to create a pthread pool at all. But this print makes it read as if pthread pool should always be used. This was not the original intent of the pool.

One way to address this might be to create Emscripten variants `emscripten_pthread_create_sync()` and `emscripten_pthread_create_async()`, where the first function would immediately return with a failure (and log an error print), if there is no Worker available to synchronously spawn a thread, and the second function would launch a thread asynchronously (taking a Worker from the pool if available, since otherwise those could pile up unused)?

Another way to address it might be to provide a build option to just mute the above message?

**Comment by sbc100** [comment]
Don't we already have a way to mute the warning?: 

```
//  - `0` - disable warnings on thread pool exhaustion                           
//  - `1` - enable warnings on thread pool exhaustion (default)                     
//  - `2` - make thread pool exhaustion a hard error                             
// [link]                                                                           
var PTHREAD_POOL_SIZE_STRICT = 1;     
```

@RReverser who added this warning/error

**Comment by RReverser** [comment]
Yeah this separate setting was intentionally added for that less common case of users creating threads asynchronously so that they could say "I know what I'm doing, I promise to create threads & wait on them in such a way that I won't deadlock myself".

**Comment by RReverser** [comment]
> So we do not want to create a pthread pool at all. But this print makes it read as if pthread pool should always be used.

FWIW I think technically it still creates & uses pool under the hood. Those extra threads would be returned to the pool when they finish their current callback (although, depending on the app, it might not happen until your app has exited anyway).
